examples/todo/src/main.rs

```diff
@@ -54,20 +60,22 @@ fn new(todo_form: Form<Todo>, conn: DbConn) -> Flash<Redirect> {
 
 #[put("/<id>")]
 fn toggle(id: i32, conn: DbConn) -> Result<Redirect, Template> {
-    if Task::toggle_with_id(id, &conn) {
-        Ok(Redirect::to("/"))
-    } else {
-        Err(Template::render("index", &Context::err(&conn, "Couldn't toggle task.")))
-    }
+    Task::toggle_with_id(id, &conn).map_err(|e| Template::render(
```

Comment:
I find this (and `delete`) easier to read as a match than with `?`, since it's the only expression:

```rust
match Task::toggle_with_id(id, &conn) {
    Ok(()) => Ok(Redirect::to("/")),
    Err(e) => Err(Template::render("index", &Context::err(&conn, format!("Couldn't toggle task: {}", e)))),
}
```

Reply:
I always think `Ok(_) => Ok(` is a little weird, but I understand where you're coming from, this is definitely easier to read for most people